2016/proposals/stolliker-ryan.md

```diff
@@ -0,0 +1,96 @@
+# Ryan Stolliker: Result Aggregation Server for Software Carpentry
+
+## Abstract
+
+During Software Carpentry's workshops which teach computing skills to the greater scientific community, learners use Python scripts to test that they have installed the programming environment and other dependencies correctly. Currently, this information is known only to the participants and cannot be used or tracked by the organizers. Thus, the goal of this project is to enable the collection of this installation data and to store it for future use and analysis.
+
+## Technical Details
+
+###Database
+
+SQLite will be used as the database to store the results of the installation scripts. SQLAlchemy will be used to manage the database and all Python interaction with the database. If database migration is needed in the future, Alembic will be used.
+
+### Server
+
+All server management will be written in Python. The server itself will be implemented using [bottle](http://bottlepy.org/docs/dev/index.html), a simple web framework with functionality to parse forms and JSON requests and responses. 
+
+####Interactions
+
+* `/first/`
+  * The first installation script will send its data here in JSON for forms format. The server will send back a 200 response.
+* `/second/`
+  * The second installation script will send its data here in JSON for forms format. The server will send back a 200 response.
+* `/data/`
+  * A request to this URL will recieve JSON from the server which contains information from the database for analysis in some future application.
+
+### Updates to Test Scripts
+
+The existing installation testing scripts, already written in Python, will be updated to send data to the server. Because the point of the scripts is to determine whether Python is configured correctly, the changes should not rely on any third party libraries and be compatible with both Python 2 and 3 so that even if a system fails a test, it will still be able to send the results. Data will be sent using the http.client and urllib.parse libraries. System data will be collected using the platform library, which is included in Python's standard library, specifically, 
```

Comment:
You should expand. Apart from my comment on database schema, you should expand on what exactly you want to gather from the installation scripts; I know you want to collect platform details, but what about test outcomes? We should be able to get insight into what was installed on participants' computer, what was working etc.

Part of your "Proposed Schema" section should be turned into textual description of what will be collected, and then you should include more technical (ie. SQL code or Python SQLAlchemy model or some UML graph) schema.